src/config/mongoose_config_spec.erl

```diff
@@ -282,17 +282,18 @@ xmpp_listener_common() ->
     #section{items = #{<<"backlog">> => #option{type = integer,
                                                 validate = non_negative},
                        <<"proxy_protocol">> => #option{type = boolean},
-                       <<"hibernate_after">> => #option{type = integer,
+                       <<"hibernate_after">> => #option{type = int_or_infinity,
                                                         validate = non_negative},
                        <<"max_stanza_size">> => #option{type = int_or_infinity,
-                                                        validate = positive},
+                                                        validate = positive,
+                                                        process = fun process_infinity_as_zero/1},
```

Comment:
We decided to use exported functions here, i.e. `fun ?MODULE:...` to be able to see the function names when you are debugging the config specs.